ignite/pkg/protoanalysis/protoutil/creator.go

```diff
@@ -0,0 +1,651 @@
+// Wrap proto structs to allow easier creation, protobuf lang is small enough
+// to easily allow this.package protoutil
+package protoutil
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/emicklei/proto"
+)
+
+// TODO: Can also support comments/inline comments? -- Probably, formatting is currently
+// flaky with how it prints them, though.
+
+// Create a new Literal:
+//
+// // true
+// l := NewLiteral("true")
+//
+// // 1
+// l := NewLiteral("1")
+//
+// // "foo"
+// l := NewLiteral("foo")
+//
+// Currently doesn't support creating compound literals (arrays/maps).
+func NewLiteral(lit string) *proto.Literal {
+	return &proto.Literal{
+		Source:   lit,
+		IsString: isString(lit),
+	}
+}
+
+// ImportSpec holds information relevant to the import statement.
+type ImportSpec struct {
+	path string
+	kind string
+}
+
+// Type alias for a callable accepting an ImportSpec.
+type ImportSpecOpts func(i *ImportSpec)
```

Comment:
It would be good that types for options end in "Option" instead of "Opts"
```suggestion
type ImportSpecOption func(i *ImportSpec)
```

Same for other cases within `protoanalisys`.

Reply:
Done but went with the plural version `*SpecOptions`